src/module/actor/character/document.ts

```diff
@@ -1316,10 +1322,15 @@ class CharacterPF2e<TParent extends TokenDocumentPF2e | null = TokenDocumentPF2e
             const hasTwoHandTrait = traitsArray.some((t) => t.startsWith("two-hand"));
             const { usage } = weapon.system;
             const canWield2H = (usage.type === "held" && usage.hands === 2) || hasFatalAimTrait || hasTwoHandTrait;
+            const { handsHeld } = weapon;
 
             switch (weapon.carryType) {
                 case "held": {
-                    if (weapon.handsHeld === 2) {
+                    if (weapon.shield && !weapon.shield.isRaised) {
+                        auxiliaryActions.push(new WeaponAuxiliaryAction({ weapon, action: "RaiseAShield" }));
+                    }
+
+                    if (handsHeld === 2) {
```

Comment:
I think you can keep this to weapon.handsHeld, since the var isn't used anywhere else? Should keep it clearer hopefully.